src/providers/ProfileModalProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useReducer } from 'react';
+
+import { User } from '../types';
+import createCtx from '../utils/createCtx';
+
+interface Context {
+  state: State;
+  showModal: (user: Partial<User>, deleteMode: boolean) => void;
+  // setUser: (user: User) => void;
+  // showAddBtn: (deleteMode: boolean) => void;
+  // open: () => void;
+}
+const [useCtx, Provider] = createCtx<Context>();
+
+export enum ActionType {
```

Comment:
Everything looks great. but I have a one question here

I guess you are using enum here for adding more Actions in the future
If my guess is right, just so you know, below pattern will give us better typing experience
```typescript
type Action =
  | { type: "SET_ONE"; payload: string }
  | { type: "SET_TWO"; payload: number };

export function reducer(state: AppState, action: Action): State {
  switch (action.type) {
    case "SET_ONE":
      return {
        ...state,
        one: action.payload // `payload` is string
      };
    case "SET_TWO":
      return {
        ...state,
        two: action.payload // `payload` is number
      };
    default:
      return state;
  }
}
```
(I am not urging you to fix your code but just giving you an idea in the future! coz my guess might be wrong! 😄)

Reply:
I'm little confused here since arguing about `enum` looks different.
![image](https://user-images.githubusercontent.com/27461460/68675233-cff0f900-059a-11ea-8e17-c49ac591e09b.png)

It looks like you'd like to switch `interface` to `type` as shown above. Is this correct?